src/actions/SignupAction.js

```diff
@@ -0,0 +1,54 @@
+import axios from 'axios';
+import { SIGNUP_PENDING, SIGNUP_SUCCESS, SIGNUP_FAIL } from './types';
+
+const appUrl = 'https://caret-bn-backend-staging.herokuapp.com/api/v1';
+
+export const signupAction = (userData) => (dispatch) => {
+  dispatch({
+    type: SIGNUP_PENDING,
+    payload: true,
+  });
+  axios.post(`${appUrl}/users/register`, userData)
+    .then((response) => {
+      dispatch({
+        type: SIGNUP_PENDING,
+        payload: false,
+      });
+      dispatch({
+        type: SIGNUP_FAIL,
+        payload: null,
+      });
+      dispatch({
+        type: SIGNUP_SUCCESS,
+        payload: response,
+      });
+    })
+    .catch((error) => {
+      dispatch({
+        type: SIGNUP_PENDING,
+        payload: false,
+      });
+      dispatch({
+        type: SIGNUP_FAIL,
+        payload: error.response,
+      });
+    });
+};
+
+export const resetSignup = () => (dispatch) => {
+  dispatch({
+    type: SIGNUP_FAIL,
+    payload: null,
+  });
+  dispatch({
+    type: SIGNUP_SUCCESS,
+    payload: false,
+  });
+  dispatch({
+    type: SIGNUP_PENDING,
+    payload: false,
+  });
+};
+
+// export default { signupAction, resetSignup };
```

Comment:
Make this file to have all auth Actions. Create one type function to be used in all the auth action functions. something like this
```
const authType = (type,payload)=>{
type,
payload
}

Make signUp action cleaner with something like this
```
  axios.post(`${appUrl}/users/register`, userData)
    .then((response) => {
        dispatch(authType(SIGNUP_SUCCESS,response))
    })
    .catch((error) => {
 dispatch(authType(SIGNUP_FAIL,error.reponse))
    });
};